blockchain/blockchain.go

```diff
@@ -1676,6 +1704,93 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 	return 0, events, coalescedLogs, nil
 }
 
+// BlockSubscriptionLoop subscribes blocks from a redis server and processes them.
+// This method is only for KES nodes.
+func (bc *BlockChain) BlockSubscriptionLoop(pool *TxPool) {
+	var ch <-chan *redis.Message
+	logger.Info("subscribe blocks from redis cache")
+
+	pubSub, ok := bc.stateCache.TrieDB().TrieNodeCache().(statedb.BlockPubSub)
+	if !ok {
+		logger.Error("invalid block pub/sub configure", "trieNodeCacheConfig", bc.stateCache.TrieDB().GetTrieNodeCacheConfig())
+	}
+
+	ch = pubSub.SubscribeBlockCh()
+	if ch == nil {
+		logger.Crit("failed to create redis subscription channel")
+	}
+
+	for msg := range ch {
+		logger.Debug("msg from redis subscription channel", "msg", msg.Payload)
+
+		blockRlp, err := hexutil.Decode(msg.Payload)
+		if err != nil {
+			logger.Error("failed to decode redis subscription msg", "msg", msg.Payload)
+			continue
+		}
+
+		block := &types.Block{}
+		if err := rlp.DecodeBytes(blockRlp, block); err != nil {
+			logger.Error("failed to rlp decode block", "msg", msg.Payload, "block", string(blockRlp))
+			continue
+		}
+
+		// TODO-Klaytn-KES: implement block handling logic
```

Comment:
This PR only receives block from redis, but does not process the block, right?

Reply:
You are right. Processing logic will be implemented later as it remarked.